### ISSUE TITLE
Missing e2e Checkout with Variants

### ISSUE THREAD

**Issue by giancorderoortiz** [SAP/spartacus#5767]
In order to verify checkout with variant products does in fact work, the following e2e tests should be added: 

1. Select single variant product, add to cart, and complete checkout. 

2. Add different variants of same product to cart and complete checkout. 

3. Add n number of skus to cart, then add single product variant and complete checkout. 

4. Repeat test 3 above with guest checkout. 

for all above, confirm variants are indeed present in order via order details page in order history. 

**Video of current product details test:** 
[product-details.e2e-spec.ts.mp4.zip](https://github.com/SAP/cloud-commerce-spartacus-storefront/files/3952655/product-details.e2e-spec.ts.mp4.zip)

**Comment by giancorderoortiz** [comment]
related to #4569

**Comment by giancorderoortiz** [comment]
See https://github.com/SAP/cloud-commerce-spartacus-storefront/blob/feature/apparel-poc/projects/storefrontapp-e2e-cypress/cypress/integration/regression/product-details.e2e-spec.ts

**Comment by mirkosoft** [comment]
@KateChuen shouldn't it be the P1 (e2e is a must for DoD)?

**Comment by KateChuen** [comment]
@mirkosoft we changed it during bug triage, as a group consensus, but you've got a good point. 
@hackergil what do you think? P1 or P2?

**Comment by mirkosoft** [comment]
we can divide work to 2-3 smaller, more precise tickets so work should go faster if two people work in parallel

**Comment by bgambocjaviniar** [comment]
will be creating another PR based on `checkout-flow` instead of the PR that has `checkout-as-persistent-user` as a base.

**Comment by bgambocjaviniar** [comment]
PR: GH-7208
